server/src/main/java/org/elasticsearch/script/field/DelegateDocValuesField.java

```diff
@@ -17,26 +17,31 @@
  * A default {@link Field} to provide {@code ScriptDocValues} for fields
  * that are not supported by the script fields api.
  */
-public class DelegateDocValuesField implements DocValuesField<Object> {
+public class DelegateDocValuesField implements Field<Object>, DocValuesScriptFieldSource {
 
     private final ScriptDocValues<?> scriptDocValues;
     private final String name;
 
     public DelegateDocValuesField(ScriptDocValues<?> scriptDocValues, String name) {
         // Suppliers provided via ScriptDocValues should never be a DocValuesField
```

Comment:
This comment is out of date.

Reply:
Good catch! It's actually a refactoring issue related to the assert, so I fixed the assert.